prow/clonerefs/run_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clonerefs
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/pod-utils/clone"
+)
+
+func TestRun(t *testing.T) {
+
+	srcRoot, err := ioutil.TempDir("", "clonerefs_unittest")
+	if err != nil {
+		t.Fatalf("Error creating temp dir: %v.", err)
+	}
+	defer os.RemoveAll(srcRoot)
+
+	type cloneRec struct {
+		refs        *kube.Refs
+		root        string
+		user, email string
+		cookiePath  string
+		env         []string
+	}
+
+	var recordedClones []cloneRec
+	cloneFuncOld := cloneFunc
+	cloneFunc = func(refs *kube.Refs, root, user, email, cookiePath string, env []string) clone.Record {
```

Comment:
Why can't we change the signature of clone.Run to `refs kube.Refs`?

Reply:
Done.